src/nvim/testdir/test_ruby.vim

```diff
@@ -0,0 +1,59 @@
+" Tests for ruby interface
+
+if !has('ruby')
+  finish
+end
+
+func Test_ruby_change_buffer()
+  call setline(line('$'), ['1 line 1'])
+  ruby Vim.command("normal /^1\n")
+  ruby $curbuf.line = "1 changed line 1"
+  call assert_equal('1 changed line 1', getline('$'))
+endfunc
+
+func Test_ruby_evaluate_list()
+  call setline(line('$'), ['2 line 2'])
+  ruby Vim.command("normal /^2\n")
+  let l = ["abc", "def"]
+  ruby << EOF
+  curline = $curbuf.line_number
+  l = Vim.evaluate("l");
+  $curbuf.append(curline, l.join("\n"))
+EOF
```

Comment:
Caught exception in Test_ruby_evaluate_list(): Vim(return):RuntimeError: String cannot contain newlines @ function RunTheTest[35]..Test_ruby_evaluate_list[8]..provider#ruby#Call, line 17

Reply:
This exposes an inconsistency with `vim`, which allows newlines in `append` strings, however I still don't think it's what you want; I tried it in `vim` and `\n` is inserted as `^@` (null byte I think). I will update neovim-ruby to not throw here, but I suggest you join on another character, like `|`. Or alternatively you can use `$curbuf.lines = l`, which will replace all the lines of the buffer.